indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -7,952 +7,330 @@
-          />
-        );
-      case 'bookableHours':
-        return (
-          <FinalField
-            key={key}
-            name="bookableHours"
-            component={DailyAvailability}
-            isEqual={_.isEqual}
-            format={value => (value === null ? [] : value)}
-          />
-        );
-      case 'nonBookablePeriods':
-        return (
-          <FinalField
-            key={key}
-            name="nonbookablePeriods"
-            component={NonBookablePeriods}
-            isEqual={_.isEqual}
-            format={value => (value === null ? [] : value)}
-          />
-        );
-      case 'protectionMode':
-        return (
-          <React.Fragment key={key}>
-            <h5>{content.label}</h5>
-            <p className="field-description">
-              <Translate>
-                Restricted rooms can only be booked by users defined in the room ACL
-              </Translate>
-            </p>
-            <Form.Group>
-              <FinalRadio name="protectionMode" value="public" label={Translate.string('Public')} />
-              <FinalRadio
-                name="protectionMode"
-                value="protected"
-                label={Translate.string('Restricted')}
-              />
-            </Form.Group>
-          </React.Fragment>
-        );
-      case 'permissions': {
-        const [permissionManager, permissionInfo] = usePermissionInfo();
-        return (
-          permissionManager &&
-          permissionInfo && (
-            <FavoritesProvider key={key}>
-              {favoriteUsersController => (
-                <FinalField
-                  name="aclEntries"
-                  component={ACLField}
-                  favoriteUsersController={favoriteUsersController}
-                  label={content.label}
-                  permissions={false}
-                  readAccessAllowed={false}
-                  isEqual={_.isEqual}
-                  withGroups
-                  permissionInfo={permissionInfo}
-                  permissionManager={permissionManager}
-                />
-              )}
-            </FavoritesProvider>
-          )
-        );
+  const closeModal = useCallback(async () => {
+    setLoading(true);
+    if ((roomId || newRoomId) !== null && wasEverUpdated) {
+      await Promise.all([
+        dispatch(roomsActions.fetchRoom(roomId)),
+        dispatch(roomsActions.fetchEquipmentTypes()),
+        dispatch(roomsActions.fetchDetails(roomId, true)),
+        dispatch(userActions.fetchRoomPermissions(roomId)),
+      ]);
+    }
+    onClose(wasEverUpdated || afterCreation);
+  }, [dispatch, onClose, roomId, newRoomId, wasEverUpdated, afterCreation]);
+
+  useEffect(() => {
+    (async () => {
+      if (!isNewRoom) {
+        setLoading(true);
+        await fetchRoomData();
+        setLoading(false);
       }
+    })();
+  }, [isNewRoom, fetchRoomData]);
+
+  const formValidation = values => {
+    if (!values.latitude !== !values.longitude) {
```

Comment:
I think the negation here is bit weird. Since you just use it to cast to boolean, maybe using the more common `!!` here?

Reply:
That was my attempt of implementing an XOR without using the bitwise operator `^`. Is your suggestion `!!a !== !!b`?

Reply:
yeah.. or since you probably always have null or a value you could also go for this: `(a === null) != (b === null)`